src/ssat/utils/ParseUtils.h

```diff
@@ -95,24 +96,28 @@ static int parseInt(B& in) {
 // parse float for SSAT solving
 template<class B>
 static double parseFloat(B& in) {
-    int     val = 0;
-    int     deg = 0;
-    double  flt;
     skipWhitespace(in);
+
+    // make sure the floating point value starts with "0."
```

Comment:
Why do floating-point values have to start with `0.`?
From the name of the function `parseFloat`, we should not have such assumption.

Reply:
Yes.
Since I only fixed the overflow issue, the original intention of the code has remained the same. (i.e. it does not accept values >= 1.0 ..)

Reply:
But I can update it if you also think that's not reasonable.